test/common.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+)
+
+const (
+	AnnotationKey   = "wetigerswoop"
+	AnnotationValue = "hurra"
+)
+
+func testName() string {
+	pc := make([]uintptr, 10)
+	n := runtime.Callers(3, pc)
+	frame, _ := runtime.CallersFrames(pc[:n]).Next()
+	funks := strings.Split(frame.Function, ".")
+	funk := funks[len(funks)-1]
+
+	return funk
+}
+
+func createPrefix() string {
+	now := time.Now().UTC().Format("20060102150405")
+	test := testName()
+
+	return now + "-" + test + "/"
+}
+
+func getDefaultCreds() *clarify.Credentials {
+	creds := clarify.BasicAuthCredentials(
+		os.Getenv("CLARIFY_INTEGRATION_ID"),
+		os.Getenv("CLARIFY_PASSWORD"),
+	)
+
+	return creds
+}
+
+func jsonEncode[v any](t *testing.T, a v) {
+	enc := json.NewEncoder(os.Stdout)
+
+	enc.SetIndent("", "  ")
+
+	err := enc.Encode(a)
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+}
+
+func createQuery(prefix string) fields.ResourceQuery {
+	return fields.Query().
+		Where(fields.Comparisons{"annotations." + prefix + AnnotationKey: fields.Equal(AnnotationValue)}).
+		Limit(10)
+}
+
+func svelg[v any](a Args, fs ...func(a Args) (v, error)) {
```

Comment:
I am not familiar with this term/abbreviation. What does it mean/refer to?

Reply:
Right... It runs all functions or panics... Probably better to return an error?

Reply:
I think in general, we would want to avoid it, and just use the explicit error handling inline.

Reply:
Got some links to the pattern, so I can read about it?